### PR TITLE
add build script for intel oneapi and gcc compiler separately

### PR DESCRIPTION
**Pull request by xyuan**
**Description**
add build env script for intel oneAPI build and gcc build of Pace with optimization options for loop unroll, and array prefetch.

please note that these options are needed to be incorported in the regression test and container build, with the purpose of supporting various supercomputers in the future. 

Fixes # (issue)
If this is a hotfix to a released version, please specify it.

**How Has This Been Tested?**
The options for Intel compiler are tested using 
Intel(R) oneAPI DPC++/C++ Compiler for applications running on Intel(R) 64, Version 2024.1.0 Build 20240308

and the gcc compiler was tesed using
gcc 13.2

**Checklist:**
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] New check tests, if applicable, are included

### CODE REVIEW

File: examples/build_scripts/build_gaea_c5_gcc.sh
Comment (fmalatino) [inline examples/build_scripts/build_gaea_c5_gcc.sh:1]: A couple of questions:

- Do we want to continue to suport python 3.8?
- Should we allow the user to pass the desired name for the conda environment to be used, or do we want to keep it as `my_name` in general?

Comment (xyuan) [reply examples/build_scripts/build_gaea_c5_gcc.sh:1]: nope, let me test python 3.11.9, and update the script

**Review by fmalatino — APPROVED** [review]
lgtm

**Comment by FlorianDeconinck** [comment]
This requires a new GT4Py

**Comment by fmalatino** [comment]
> This requires a new GT4Py

@xyuan Does this PR depend on your [PR 1552](https://github.com/GridTools/gt4py/pull/1552) in GT4Py?

**Comment by xyuan** [comment]
> > This requires a new GT4Py
> 
> @xyuan Does this PR depend on your [PR 1552](https://github.com/GridTools/gt4py/pull/1552) in GT4Py?

yes, but it is still safe to merge it to pace

File: examples/build_scripts/build_gaea_c5_gcc.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+set -e -x
+
+# module load necessary system software
+module rm PrgEnv-intel
+module load PrgEnv-gnu
+module rm gcc
+module load gcc/12.2.0
+module load boost/1.79.0
+module load python/3.9
```
Comment (oelbert) [inline examples/build_scripts/build_gaea_c5_gcc.sh]: Should we update the python version here too?

**Review by oelbert — COMMENTED** [review]
Small q about the python module import on C5, otherwise LGTM

**Review by fmalatino — CHANGES_REQUESTED** [review]
Approved pending changes to `examples/build_scripts/build_gaea_c5_gcc.sh` to update python version to 3.11

File: examples/build_scripts/build_gaea_c5_gcc.sh (hunk 1 above)
Comment (xyuan) [reply examples/build_scripts/build_gaea_c5_gcc.sh]: yes, updated to the python 3.11.7

**Review by fmalatino — APPROVED** [review]
lgtm

**Review by oelbert — APPROVED** [review]
Assuming the gt4py version is ok we're 👍

**Pull request merged**